java/ql/src/Security/CWE/CWE-1104/MavenPomDependsOnBintray.ql

```diff
@@ -0,0 +1,33 @@
+/**
+ * @name Depending upon JCenter/Bintray as an artifact repository
+ * @description JCenter & Bintray are deprecated
+ * @kind problem
+ * @problem.severity error
+ * @precision very-high
+ * @id java/maven/dependency-upon-bintray
+ * @tags security
+ *       external/cwe/cwe-1104
+ */
+
+import java
+import semmle.code.xml.MavenPom
+
+private class DeclaredRepository extends PomElement {
+  DeclaredRepository() {
+    this.getName() = "repository" or
+    this.getName() = "snapshotRepository" or
+    this.getName() = "pluginRepository"
```

Comment:
You could use a [set literal expression](https://codeql.github.com/docs/ql-language-reference/expressions/#set-literal-expressions) here.

Reply:
How does that work for these kinds of checks?

Reply:
You can use it (as far as I know) everywhere where you would normally use expressions. Here it would be:
```ql
this.getName() = ["repository", "snapshotRepository", "pluginRepository"]
```

Reply:
Oh, awesome! Thanks!